tools/query_breakdown/src/main/java/com/google/bigquery/QueryBreakdown.java

```diff
@@ -62,23 +64,51 @@ public void run(String originalQuery, String outputFile, int errorLimit,
       stack.push(current);
       current = current.getParent();
     }
-
-    // we then pop the stack and output results one by one
-    while (!stack.empty()) {
-      current = stack.pop();
-      if (current.getErrorHandlingType().equals("DELETION")) {
-        // print out the result
-        System.out.println(String.format("Unparseable portion: Start Line %1$s, End Line %2$s, "
-                + "Start Column %3$s, End Column %4$s, %5$s", current.getStartLine(),
-            current.getEndLine(), current.getStartColumn(), current.getEndColumn(),
-            current.getErrorHandlingType()));
+    
+    if (jsonOutput) {
+      JSONArray jsonArray = new JSONArray();
+      while(!stack.empty()) {
+        current = stack.pop();
+        JSONObject errorPosition = new JSONObject();
+        errorPosition.put("startLine", current.getStartLine());
```

Comment:
You should be able to simplify this by using a library like Jackson (https://en.wikipedia.org/wiki/Jackson_(API)). You can choose to do that in a future PR.

Reply:
Got it, yup. Right now I might still add more to the JSON schema so I will address this in a future PR!